src/mesh/mesh_tools.C

```diff
@@ -366,9 +366,7 @@ void MeshTools::find_boundary_nodes (const MeshBase & mesh,
     for (auto s : elem->side_index_range())
       if (elem->neighbor_ptr(s) == nullptr) // on the boundary
         {
-          std::unique_ptr<const Elem> side = elem->build_side_ptr(s);
```

Comment:
What?  How did we never get an unused variable warning for this?  Is the non-trivial destructor enough for the compiler to have to give up and say "maybe they're doing some fancy RAII thing"?

Reply:
Right?! I had the same thought!